packages/git/src/browser/diff/git-diff-service.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { GitLogOptions } from './git-diff-model';
+import { injectable, inject } from "inversify";
+import { GitRepositoryProvider } from '../git-repository-provider';
+import { Git, Repository, GitFileChange, GitUtils } from '../../common';
+import { FileUri } from '@theia/core/lib/node/file-uri';
+import * as Path from 'path';
+
+@injectable()
+export class GitDiffService {
+
+    constructor(
+        @inject(Git) protected readonly git: Git,
+        @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider
+    ) { }
+
+    async getDiff(repository: Repository, options: GitLogOptions): Promise<GitFileChange[]> {
+        const range = this.getRangeArg(options);
+        const args = ['diff', '--name-status', range];
+        if (options.fileUri) {
+            args.push(...['--', GitUtils.getRepositoryRelativePath(repository, options.fileUri)]);
+        }
+        const changes: GitFileChange[] = [];
+        (await this.git.exec(repository, args)).stdout.split('\0').map(line => line.match(/\S+/g) || []).forEach(fragments => {
+            for (let i = 0; i < fragments.length; i = i + 2) {
+                const status = GitUtils.mapStatus(fragments[i]);
+                const uri = this.toUri(repository, fragments[i + 1]);
+                changes.push({
+                    uri,
+                    status
+                });
+            }
+        });
+        return changes;
+    }
+
+    toUri(repository: Repository, pathSegment: undefined): undefined;
```

Comment:
I would reduce the visibility of these and `getRangeArg` too to at least protected.